### PR TITLE
Add switchconfig type

### PR DESCRIPTION
**Pull request by jerearista**

### CODE REVIEW

File: lib/puppet/provider/eos_switchconfig/default.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  end
+
+  def staging_file=(value)
+    @property_flush[:staging_file] = value
+  end
+
+  def flush
+    # Write staging_file to flash
+    open('/mnt/flash/' + @property_flush[:staging_file], 'w') do |f|
+      f.puts @property_flush[:content]
```
Comment (jerearista) [inline lib/puppet/provider/eos_switchconfig/default.rb]: Check for failure... out of space, etc?

File: lib/puppet/provider/eos_switchconfig/default.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+  def self.instances
+    # Get the current value
+    conf = node.get_config(config: 'running-config', as_string: true)
+    # Remove comment lines and the end statement from conf
+    # The end statement is removed to allow comparison with user
+    # specified switch configs that do not have the end statement.
+    conf_arr = []
+    conf.each_line do |line|
+      #next if line.start_with?('!') || line.start_with?('end')
```
Comment (jerearista) [inline lib/puppet/provider/eos_switchconfig/default.rb]: remove

File: lib/puppet/type/eos_switchconfig.rb [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        '#{desired_lines.join("\n")}'"
+    end
+  end
+
+  newproperty(:staging_file) do
+    desc <<-EOS
+      The staging_file is the actual file which will be managed on flash:
+      on the switch before running 'configure replace'.
+
+      The default value is 'startup-config'.
```
Comment (jerearista) [inline lib/puppet/type/eos_switchconfig.rb]: Change default to /mnt/flash/puppet-config

File: lib/puppet/type/eos_switchconfig.rb [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+Puppet::Type.newtype(:eos_switchconfig) do
+  @doc = <<-EOS
+    Manage the complete EOS config as a file
+
+    Use files, templates, or concatenated files/template blocks to build
+    and manage the entire EOS configuration as a single object.  By default,
+    changes will be written to flash:startup-config then the 'configure
```
Comment (devrobo) [inline lib/puppet/type/eos_switchconfig.rb]: Is the default now puppet-config?

Comment (jerearista) [inline lib/puppet/type/eos_switchconfig.rb]: Yes.  Resolved.

**Pull request closed without merging**